app/controllers/api/base_controller.rb

```diff
@@ -10,4 +10,26 @@ def find_rubygem_by_name
     return if @rubygem || @gem_name == WebHook::GLOBAL_PATTERN
     render text: "This gem could not be found", status: :not_found
   end
+
+  def validate_gem_and_version
+    if !@rubygem.hosted?
+      render text: t(:this_rubygem_could_not_be_found),
+             status: :not_found
+    elsif !@rubygem.owned_by?(current_user)
+      render text: "You do not have permission to mark this gem as vulnerable.",
```

Comment:
Won't this say `"You do not have permission to mark this gem as vulnerable."` when user doesn't have permission to yank gem? May be it can use different wording.